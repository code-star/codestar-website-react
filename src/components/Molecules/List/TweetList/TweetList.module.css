.tweetList {
	max-width: 500px;
}
