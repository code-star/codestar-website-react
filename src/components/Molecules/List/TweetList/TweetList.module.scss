.card {
  max-width: 345px;
  margin: 1em;
  display: flex;
  flex-direction: column;
}

.media {
  height: 0;
  padding-top: 56.25%; /* 16:9 */
}

a {
  color: black;

  &:hover {
    text-decoration: none;
  }

  .prefix {
    color: #888;
  }
}
