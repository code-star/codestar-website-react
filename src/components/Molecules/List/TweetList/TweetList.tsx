import React from 'react';
import { withStyles } from '@material-ui/core/styles';
import {
  Card,
  CardActionArea,
  CardContent,
  Typography,
  CardActions,
  Button,
} from '@material-ui/core';

interface ITweetListProps {
  tweets: any[];
}

const styles = {
  card: {
    maxWidth: 345,
  },
  media: {
    height: 140,
  },
};

const TweetList = ({ tweets }: ITweetListProps) => (
  <Card>
    <CardActionArea>
      <CardContent>
        <Typography gutterBottom variant="headline" component="h2">
          {tweets[0].user.name} tweets ...
        </Typography>
        {tweets.map(tweet => {
          return (
            <div key={tweet.id} className="p-2">
              <Typography component="p">
                <a
                  href={`https://twitter.com/${tweet.user.screen_name}/status/${
                    tweet.id_str
                  }`}
                >
                  <span className="text-info">{`${new Date(
                    tweet.created_at
                  ).toDateString()} `}</span>
                  {`${tweet.text}`}
                </a>
              </Typography>
            </div>
          );
        })}
      </CardContent>
    </CardActionArea>
    <CardActions>
      <Button
        size="small"
        color="primary"
        href={`https://twitter.com/${tweets[0].user.screen_name}?lang=en`}
      >
        Read More
      </Button>
    </CardActions>
  </Card>
);

export default withStyles(styles)(TweetList);
