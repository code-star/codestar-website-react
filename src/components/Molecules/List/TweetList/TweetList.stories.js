import React from 'react';
import { storiesOf } from '@storybook/react';
import { withKnobs } from '@storybook/addon-knobs';
import { Button } from '@material-ui/core';
import TweetList from './TweetList.tsx';
import tweets from '../../../../../public/mock/get-recent-tweets';

storiesOf('components/Molecules/List', module)
  .addDecorator(withKnobs)
  .add('TweetList', () => (
    <TweetList
      tweets={tweets.slice(0, 5)}
      eventDate="December 13, 2018"
      eventImage="https://secure.meetupstatic.com/photos/event/3/5/b/b/600_473173755.jpeg"
      eventName="Data Oriented Design with Maxim Zaks"
    >
      <Button size="small" color="primary" href="#">
        Sign Up
      </Button>
    </TweetList>
  ));
