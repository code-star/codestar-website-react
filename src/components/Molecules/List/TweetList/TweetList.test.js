import React from 'react';
import { shallow } from 'enzyme';

import TweetList from './TweetList';

const getComponent = tweets => <TweetList tweets={tweets} />;

const someTweets = [
	{
		user: {
			name: 'someAwesomeName',
			screen_name: 'someAwesomeScreenName',
		},
		created_at: 'Fri Aug 31 13:51:31 +0000 2018',
		text: 'some awesome text',
		favorite_count: 3,
		retweet_count: 5,
	},
	{
		user: {
			name: 'someOtherAwesomeName',
			screen_name: 'someOtherAwesomeScreenName',
		},
		created_at: 'Sun Sep 02 12:40:19 +0000 2018',
		text: 'some other awesome text',
		favorite_count: 1,
		retweet_count: 2,
	},
];

const renderShallow = () => {
	return shallow(getComponent(someTweets));
};

describe('<TweetList />', () => {
	let wrapper;

	describe('Instance', () => {
		test('must be an instance of TweetList', () => {
			wrapper = renderShallow();
			expect(wrapper.find('TweetList')).toBeTruthy();
		});
	});

	describe('Snaphot', () => {
		test('must match some tweet list', () => {
			expect(global.renderToJSON(getComponent(someTweets))).toMatchSnapshot();
		});
	});
});
