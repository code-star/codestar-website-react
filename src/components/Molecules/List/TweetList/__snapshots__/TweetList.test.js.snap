// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TweetList /> Snaphot must match some tweet list 1`] = `
<div
  className="MuiPaper-root-4 MuiPaper-elevation1-7 MuiPaper-rounded-5 MuiCard-root-3"
>
  <button
    className="MuiButtonBase-root-34 MuiCardActionArea-root-31"
    disabled={undefined}
    onBlur={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex="0"
    type="button"
  >
    <div
      className="MuiCardContent-root-37"
    >
      <h2
        className="MuiTypography-root-38 MuiTypography-headline-43 MuiTypography-gutterBottom-56"
      >
        someAwesomeName
         tweets ...
      </h2>
      <div
        className="p-2"
      >
        <p
          className="MuiTypography-root-38 MuiTypography-body1-47"
        >
          Fri Aug 31 2018 some awesome text
        </p>
      </div>
      <div
        className="p-2"
      >
        <p
          className="MuiTypography-root-38 MuiTypography-body1-47"
        >
          Sun Sep 02 2018 some other awesome text
        </p>
      </div>
    </div>
    <span
      className="MuiCardActionArea-focusHighlight-33"
    />
    <span
      className="MuiTouchRipple-root-64"
    />
  </button>
  <div
    className="MuiCardActions-root-71"
  >
    <a
      className="MuiButtonBase-root-34 MuiButton-root-73 MuiButton-text-75 MuiButton-textPrimary-76 MuiButton-flat-78 MuiButton-flatPrimary-79 MuiButton-sizeSmall-96 MuiCardActions-action-72"
      href="https://twitter.com/someAwesomeScreenName?lang=en"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex="0"
    >
      <span
        className="MuiButton-label-74"
      >
        Read More
      </span>
      <span
        className="MuiTouchRipple-root-64"
      />
    </a>
  </div>
</div>
`;
